Test Report CI action is broken

The CI action stopped working on Oct 17 (https://github.com/dask/distributed/actions/runs/6544286691) and the test report hasn't been updated since.

```
Traceback (most recent call last):
Getting list of workflow runs...
Found 274 workflow runs
Fetching artifact listing for the 30 most recent workflow runs
Downloading and parsing 687 artifacts...
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 508, in <module>
    main()
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 392, in main
    dfs = list(
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 344, in download_and_parse_artifacts
    jobs_df = get_jobs(r, session=session)
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line [14](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:15)9, in get_jobs
    raise ValueError(f"Job names must have 4 components:\n{name_components!r}")
ValueError: Job names must have 4 components:
Empty DataFrame
Columns: []
Index: []
20... 40... 60... 80... 100... 120... 140... [16](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:17)0... [18](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:19)0... [20](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:21)0... [22](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:23)0... [24](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:25)0... [26](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:27)0... [28](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:29)0... [30](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:31)0... 320... 340... 360... 380... 400... 420... 440... 460... 480... 500... 520... 540... 560... 580... 600... 620... 640... 
Error: Process completed with exit code 1.
```

--- a/.github/workflows/tests.yaml
+++ b/.github/workflows/tests.yaml
@@ -17,6 +17,7 @@ jobs:
     # Do not run the schedule job on forks
     if: github.repository == 'dask/distributed' || github.event_name != 'schedule'
     runs-on: ${{ matrix.os }}
+    # If you change the pattern of this name, please adjust the scripts/test_report.py parsing accordingly
     name: ${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.label }}-${{ matrix.partition }}
     timeout-minutes: 120
 

--- a/continuous_integration/scripts/test_report.py
+++ b/continuous_integration/scripts/test_report.py
@@ -140,14 +140,10 @@ def get_jobs(run, session):
     # Somehow the job ID is not part of the workflow schema and we have no other way to later join
     # this to the JXML results.
 
-    name_components = (
-        df_jobs.name.str.extract(r"test \((.+)\)", expand=False)
-        .dropna()
-        .str.split(", ", expand=True)
-    )
-    if len(name_components.columns) != 4:
-        raise ValueError(f"Job names must have 4 components:\n{name_components!r}")
-    name_components.columns = ["OS", "environment", "label", "partition"]
+    name_components = df_jobs.name.str.split("-", expand=True).dropna()
+    if len(name_components.columns) != 5:
+        raise ValueError(f"Job names must have 5 components:\n{name_components!r}")
+    name_components.columns = ["OS", "OS version", "environment", "label", "partition"]
 
     # See `Set $TEST_ID` step in `tests.yaml`
     name_components["partition"] = name_components.partition.str.replace(" ", "")

